en/02_Developer_Guides/02_Controllers/07_CMS_JSON_APIs.md

```diff
@@ -0,0 +1,224 @@
+
+The reason for not implementing PUT or PATCH is because `FormSchema` and react `FormBuilder` should be used whenever you need to render a form that uses the standard set of react form fields. `FormSchema` and `FormBuilder` will handle data submission, data validation and showing validation errors on the frontend.
+
+`FormSchema` diverges from REST principles as it utilises a combination of JSON to retrieve the `FormSchema` and submitting data using `application/x-www-form-urlencoded` POST requests. This method has proven effective, and there are currently no plans to alter its functionality.
+
+Because of this you should generally avoid updating large parts of a DataObject with either a POST, PUT or PATCH requests, since `FormSchema` can handle that for you. We recommend only use POST requests to submit small amounts of data such as creating a new record without any data or updating sort order fields.
+
+## Creating a controller
+
+Create a subclass of [`LeftAndMain`](api:SilverStripe\Admin\LeftAndMain). This ensures that users must be logged in to the admin interface to access the endpoint. Additionally, it provides access to the methods [`LeftAndMain::jsonSuccess()`](api:SilverStripe\Admin\LeftAndMain::jsonSuccess()) and [`LeftAndMain::jsonError()`](api:SilverStripe\Admin\LeftAndMain::jsonError()).
+
+[warning]
+To enhance security, do not create a direct subclass of [`Controller`](api:SilverStripe\Control\Controller) routed using YAML on the `/admin` route. This practice is strongly discouraged as it circumvents the requirement to log in to the CMS to access the endpoints. At best you'd be re-implementing logic that already exists.
+[/warning]
+
+When naming this class, it's best practice to add a "Controller" suffix to this class, for instance name it `MySomethingController`.
+
+Define the URL segment of your controller using the [`url_segment`](api:SilverStripe\Admin\LeftAndMain->url_segment) configuration property. For example `private static string $url_segment = 'my-segment';`. For small optional modules, this may typically be the composer name of the module, for instance "linkfield".
+
+se the [`required_permission_codes`](api:SilverStripe\Admin\LeftAndMain->required_permission_codes) configuration property to declare what permissions are required to access endpoints on the controller. For example `private static string $required_permission_codes = 'CMS_ACCESS_CMSMain';`.
+
+See [user permissions](/developer_guides/security/permissions/) for more information about declaring permissions.
+
+As this is a subclass of `LeftAndMain`, it automatically gets added to the CMS menu. To remove it from the CMS menu, create a `_config.php` in the module (if it doesn't already exist) and remove the controller from the menu like so:
+
+```php
+use App\Controllers\MySomethingController;
+use SilverStripe\Admin\CMSMenu;
+
+CMSMenu::remove_menu_class(MySomethingController::class);
+```
+
+## Handling requests with $url_handlers
```

Comment:
```suggestion
## Handling requests with `$url_handlers`
```
Sorry, should have caught this one first time around.

Reply:
Updated